### PR TITLE
Inject user into controller

### PR DESCRIPTION
**Pull request by vorfolomey**
#1374

### CODE REVIEW

**Review by Yura09 — APPROVED** [review]

File: core/src/main/java/greencity/annotations/CurrentUser.java [hunk 1]
```diff
@@ -0,0 +1,11 @@
+package greencity.annotations;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+@Retention(RetentionPolicy.RUNTIME)
+@Target(ElementType.PARAMETER)
+public @interface CurrentUser {
```
Comment (antonbakalets) [inline core/src/main/java/greencity/annotations/CurrentUser.java:10]: Need a javaDoc here with 
`{@link UserArgumentResolver}`

File: core/src/main/java/greencity/converters/UserArgumentResolver.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+import org.springframework.web.method.support.HandlerMethodArgumentResolver;
+import org.springframework.web.method.support.ModelAndViewContainer;
+
+@Component
+@AllArgsConstructor
+public class UserArgumentResolver implements HandlerMethodArgumentResolver {
+    UserService userService;
+
+    /**
+     * Method checks if parameter is User.class and annotated CurrentUser.class.
```
Comment (antonbakalets) [inline core/src/main/java/greencity/converters/UserArgumentResolver.java:25]:
```
    /**
     * Method checks if parameter is {@link User} and is annotated with {@link CurrentUser}.
     * @param parameter method parameeter
     * @return boolean
     */
```

**Review by antonbakalets — APPROVED** [review]

File: core/src/main/java/greencity/annotations/CurrentUser.java (hunk 1 above)
Comment (vorfolomey) [reply core/src/main/java/greencity/annotations/CurrentUser.java:14]: Okay

File: core/src/main/java/greencity/converters/UserArgumentResolver.java (hunk 2 above)
Comment (vorfolomey) [reply core/src/main/java/greencity/converters/UserArgumentResolver.java]: Okay

**Review by eyouu — APPROVED** [review]

**Pull request merged**